### ISSUE TITLE
add customizable auth token path `HF_HUB_TOKEN_PATH`

### ISSUE THREAD

**Issue by stas00** [huggingface/huggingface_hub#2181]
**Is your feature request related to a problem? Please describe.**

Teams often needs to share `HF_HOME` to avoid duplication of models/datasets.

The problem is that `HF_HOME/token` is now shared as well, which means that everything is fine as long as models are public, but as soon as there is a gate (e.g. approval), then we no longer can share the token.

**Describe the solution you'd like**

add new env var `HF_HUB_TOKEN_PATH` to allow users to override the path.

**Comment by xwinxu** [comment]
+1 to this!

**Comment by julien-c** [comment]
+1

**Comment by julien-c** [comment]
anyone wants to open a quick PR to do this?

**Comment by Wauplin** [comment]
Solved in https://github.com/huggingface/huggingface_hub/pull/2185.

**Comment by stas00** [comment]
any plans to make a new release please that would include this feature?

**Comment by Wauplin** [comment]
Yep sorry for the delay. [Pre-release has been shipped](https://pypi.org/project/huggingface-hub/#history) just on Tuesday and release should happen anytime today or tomorrow. You can check the release notes there: https://huggingface.co/spaces/Wauplin/huggingface_hub/discussions/6.

**Comment by julien-c** [comment]
That's awesome!

**Comment by Wauplin** [comment]
Great to hear that! :)
